### ISSUE TITLE
allContainers and sledgehammer are not configurable with maven properties

### ISSUE THREAD

**Issue by fjavierjimenez** [fabric8io/docker-maven-plugin#502]
Hello,

the only way to configure those properties is to execute mvn -Ddocker.allContainers .... and I think It should be configurable with maven properties: 

`  <properties>
    <docker.allContainers>true</docker.allContainers>
  </properties>`

**Comment by rhuss** [comment]
Thanks for the PR. I wonder though, what your use case is, since using this will always stop any container from this image, even when started manually by `docker run`. Its ok when given for specific call to 'docker:stop', but I don't see yet how it could be useful when used as a permanent configuration option. 

would be awesome if you could elaborate a bit, where you need this as project property.

**Comment by fjavierjimenez** [comment]
It's not a project requirement but when I'm playing with docker containers, sometimes I forgot to stop then and my integration tests fail cause I forgot to stop a container and a port is already used. I just want to make sure that my integration tests work, independently if I have stopped my containers or not.

Anyway, The documentation should be consistent, in the current version the properties keepContainer, keepRunning and removeVolumes are configurable with a maven property and allContainers not. But the documentation don't say anything about this. So I think there is three posiblilities, write in the documentation that allContainers is not configurable with a maven property, make all the properties listed above configurable with maven properties or none of the properties are configurable with a maven property.

**Comment by rhuss** [comment]
You are absolutely correct, it should be at least mentioned in the docs and be consistent.

My current issue is, that I can't simply take over 'docker.allContainers' to the properties because it has no meaning without the context of 'docker:stop' (and the properties are used to configured all goals). So I would need to introduce / rename a `docker.stopAllContainers`.

btw the (undocumented) `sledgeHammer` is an concession to @jgangemi who seem to love martial associations ;-P and who contributed this feature. So it's just an easter egg ...

I probably go this way to introduce stopAllContainers, make allContainers deprecated and support it for both, system and project props with a corresponding warning in the docs that it definitely will kill all containers associated with the images in the build.

thanks for the heads up ...

**Comment by fjavierjimenez** [comment]
thanks @rhuss !, now I understands why `sledgeHammer` exist :-P

You convinced me to keep `allContainers` as it is for now. Deprecating `allContainers` should be a trivial change but if you need some help just let me know.

Should I close the issue?

**Comment by rhuss** [comment]
thx ;-)

No, lets keep it open to least as an reminder the documentation needs to be harmonized ....
